### ISSUE TITLE
Running precision landing on the FCU with landing target plugin doesn't seem to be working

### ISSUE THREAD

**Issue by canberkgurel** [mavlink/mavros#1363]
Hi all, 

I am working on a precision landing problem where I have a custom vision node ([aruco_detect](https://github.com/UbiquityRobotics/fiducials)) that runs on the companion computer (Ubuntu 16.04, ROS Kinetic) and passes the pose estimation to the [landing_target.cpp plugin](https://github.com/mavlink/mavros/blob/master/mavros_extras/src/plugins/landing_target.cpp). The message type that I'm using is PoseStamped ([geometry_msgs::PoseStamped](http://docs.ros.org/melodic/api/geometry_msgs/html/msg/PoseStamped.html)) msg. I added a few print statements into landing_target.cpp and captured a snapshot of the data that the plugin is receiving. 

When I'm testing, the drone arms, switches to the GUIDED mode, and takes off to 3m, then it flies over the aruco marker. I monitor the output of the vision node and when it starts publishing the pose estimation (a marker is being detected), the drone switches to the land mode and lands. I expect the drone to follow the landing target that is set while it's descending, which doesn't appear to be the case. It looks like it doesn't take corrective actions and instead it lands where it is.  

### MAVROS version and platform

Mavros: ?0.18.4?
ROS: Kinetic
Ubuntu: 16.04

### Autopilot type and version

[ x ] ArduPilot
[  ] PX4

Version: 3.6.11

**Comment by aydal** [comment]
@canberkgurel Were you able to solve it?

**Comment by canberkgurel** [comment]
@aydal No. But I have a few updates.

I started using the [cartesian_to_displacement](https://github.com/mavlink/mavros/blob/63183a6fee196b0620a7f7749c5ddeae5b50c669/mavros_extras/src/plugins/landing_target.cpp#L260) function in void send_landing_target. As a result the drone started oscillating when I trigger landing. This answers my Q1: the data flow is fine. 

As you can see in the GIF below, the drone starts circling above the target and the eventually fades away from the target. I still need help to solve this problem.

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/33564573/71565326-98bba600-2a7b-11ea-8415-a6ca8adb7aa6.gif)

**Comment by aydal** [comment]
Hello @canberkgurel

**Comment by khbnzr** [comment]
@aydal We are using the APM software stack and in that the mode for precision landing is "land"(after setting the proper precland enable params).

**Comment by mohammedshafeeqet** [comment]
@canberkgurel  @khabbab-nazar @m1baldwin
Were you able to see the messages in MAVlink inspector?

**Comment by canberkgurel** [comment]
@DoneMan No, not at the time. We'll re-attempt soon with more recent mavros, mavlink, and PX4-Autopilot versions.

**Comment by BrenoPimentel** [comment]
Did you guys solve it?
